### PR TITLE
[iconv] update type definition to 3.0

### PR DESCRIPTION
**Pull request by OpportunityLiu**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://github.com/bnoordhuis/node-iconv>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/iconv/index.d.ts [hunk 1]
```diff
@@ -1,37 +1,29 @@
...
+}
 
-        // copy from NodeJS.WritableStream for compatibility
-        write(buffer: Buffer|string, cb?: Function): boolean;
-        write(str: string, encoding?: string, cb?: Function): boolean;
-        end(): void;
-        end(buffer: Buffer, cb?: Function): void;
-        end(str: string, cb?: Function): void;
-        end(str: string, encoding?: string, cb?: Function): void;
+export interface Iconv extends Stream {
```
Comment (OpportunityLiu) [inline types/iconv/index.d.ts:16]: https://github.com/bnoordhuis/node-iconv/blob/b065855fb183caf756539da60fab3f32456b701b/index.js#L92

File: types/iconv/index.d.ts [hunk 2]
```diff
@@ -1,37 +1,29 @@
...
+    readonly writable: true;
```
Comment (OpportunityLiu) [inline types/iconv/index.d.ts:17]: https://github.com/bnoordhuis/node-iconv/blob/b065855fb183caf756539da60fab3f32456b701b/index.js#L47

File: types/iconv/index.d.ts [hunk 3]
```diff
@@ -1,37 +1,29 @@
...
-    interface Iconv extends NodeJS.WritableStream {
-
-        writable: boolean;
-        convert(input: string | Buffer, encoding?: string): Buffer;
-        write(input: string | Buffer, encoding?: string): boolean;
-        end(input?: string | Buffer, encoding?: string): void;
+interface Static {
+    new (fromEncoding: string, toEncoding: string): Iconv;
+    (fromEncoding: string, toEncoding: string): Iconv;
+    prototype: Iconv;
```
Comment (OpportunityLiu) [inline types/iconv/index.d.ts:13]: https://github.com/bnoordhuis/node-iconv/blob/b065855fb183caf756539da60fab3f32456b701b/index.js#L43

File: types/iconv/index.d.ts [hunk 4]
```diff
@@ -1,37 +1,29 @@
...
-        write(buffer: Buffer|string, cb?: Function): boolean;
-        write(str: string, encoding?: string, cb?: Function): boolean;
-        end(): void;
-        end(buffer: Buffer, cb?: Function): void;
-        end(str: string, cb?: Function): void;
-        end(str: string, encoding?: string, cb?: Function): void;
+export interface Iconv extends Stream {
+    readonly writable: true;
+    convert(input: string, encoding?: BufferEncoding): Buffer;
+    convert(input: Buffer): Buffer;
```
Comment (OpportunityLiu) [inline types/iconv/index.d.ts:19]: https://github.com/bnoordhuis/node-iconv/blob/b065855fb183caf756539da60fab3f32456b701b/index.js#L59-L64

File: types/iconv/index.d.ts [hunk 5]
```diff
@@ -1,37 +1,29 @@
...
+    write(input: string, encoding?: BufferEncoding): boolean;
+    write(input: Buffer): boolean;
```
Comment (OpportunityLiu) [inline types/iconv/index.d.ts:21]: https://github.com/bnoordhuis/node-iconv/blob/b065855fb183caf756539da60fab3f32456b701b/index.js#L66-L82

File: types/iconv/index.d.ts [hunk 6]
```diff
@@ -1,37 +1,29 @@
...
 
-        // copy from stream.Stream
-        pipe<T extends NodeJS.WritableStream>(destination: T, options?: { end?: boolean; }): T;
-    }
+    end(input: string, encoding?: BufferEncoding): void;
+    end(input?: Buffer): void;
```
Comment (OpportunityLiu) [inline types/iconv/index.d.ts:24]: https://github.com/bnoordhuis/node-iconv/blob/b065855fb183caf756539da60fab3f32456b701b/index.js#L84-L90

File: types/iconv/index.d.ts [hunk 7]
```diff
@@ -1,37 +1,29 @@
...
 }
 
-declare var Iconv: Iconv.Static;
-export = Iconv;
+export const Iconv: Static;
+
+export {};
```
Comment (OpportunityLiu) [inline types/iconv/index.d.ts:29]: https://github.com/bnoordhuis/node-iconv/blob/b065855fb183caf756539da60fab3f32456b701b/index.js#L19

**Review by orta — APPROVED** [review]
Looks good to me in comparison to the JS, thanks for providing all the inline links

**Comment by OpportunityLiu** [comment]
Ready to merge

**Pull request merged**